shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/wal/OpenGaussLogicalReplication.java

```diff
@@ -52,7 +57,28 @@ public Connection createConnection(final StandardPipelineDataSourceConfiguration
         PGProperty.ASSUME_MIN_SERVER_VERSION.set(props, "9.4");
         PGProperty.REPLICATION.set(props, "database");
         PGProperty.PREFER_QUERY_MODE.set(props, "simple");
-        return DriverManager.getConnection(jdbcConfig.getJdbcUrl(), props);
+        try {
+            return DriverManager.getConnection(jdbcConfig.getJdbcUrl(), props);
+        } catch (final SQLException ex) {
+            if (failedBecauseOfNonHAPort(ex)) {
+                log.info("Failed to connect to openGauss caused by: {} - {}. Try connecting to HA port.", ex.getSQLState(), ex.getMessage());
+                return tryConnectingToHAPort(jdbcConfig.getJdbcUrl(), props);
+            }
+            throw ex;
+        }
+    }
+    
+    private boolean failedBecauseOfNonHAPort(final SQLException ex) {
+        return ex.getMessage().contains(HA_PORT_ERROR_MESSAGE_KEY);
```

Comment:
Is `ex.getMessage()` might be null?

Reply:
Message will usually be present if the error was returned by database.